[import-w3c-tests] Fix reftest path resolution with -s

gsnedders opened the pull request:
#### c2f1b3775319389cbd947580b2af62d88a19de9a
<pre>
[import-w3c-tests] Fix reftest path resolution with -s
<a href="https://bugs.webkit.org/show_bug.cgi?id=257674">https://bugs.webkit.org/show_bug.cgi?id=257674</a>
rdar://problem/110201843

Reviewed by Jonathan Bedard.

Previously, bf94a2f96abf implemented this with differing behaviour
depending on whether `-s` was provided; if `-s` wasn&apos;t provided then
absolute reftest links were resolved relative to the WPT repo root, but
if `-s` was provided then they were resolved relative to the parent of
the WPT repo root.

This difference obviously breaks workflows where `-s` is being provided
to avoid having to re-clone WPT, and it&apos;s unclear what the motivation
for the difference in behaviour was.

We now behave the same in both cases, thus passing `-s` to avoid
re-cloning WPT still leads to the correct behaviour.

* Tools/Scripts/webkitpy/w3c/test_importer.py:
(TestImporter._source_root_directory_for_path):
* Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:

Canonical link: <a href="https://commits.webkit.org/267087@main">https://commits.webkit.org/267087@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8ed8ca73124bbfbeeddd006a146913c747f174ec

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/15327 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/15630 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/15992 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/17083 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/14387 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/15470 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/18147 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/15730 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/16986 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/15510 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/15937 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/13030 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/17817 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/15431 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/13215 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/13825 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/20773 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/14297 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/13992 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/17248 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/14568 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/12341 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/13838 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/18185 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1902 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/14401 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk reviewed (commented):
Non-reviewer looks good.

gsnedders committed:
[import-w3c-tests] Fix reftest path resolution with -s
https://bugs.webkit.org/show_bug.cgi?id=257674
rdar://problem/110201843

Reviewed by Jonathan Bedard.

Previously, bf94a2f96abf implemented this with differing behaviour
depending on whether `-s` was provided; if `-s` wasn't provided then
absolute reftest links were resolved relative to the WPT repo root, but
if `-s` was provided then they were resolved relative to the parent of
the WPT repo root.

This difference obviously breaks workflows where `-s` is being provided
to avoid having to re-clone WPT, and it's unclear what the motivation
for the difference in behaviour was.

We now behave the same in both cases, thus passing `-s` to avoid
re-cloning WPT still leads to the correct behaviour.

* Tools/Scripts/webkitpy/w3c/test_importer.py:
(TestImporter._source_root_directory_for_path):
* Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:

Canonical link: https://commits.webkit.org/267087@main
